crates/transaction-emitter-lib/src/emitter/mod.rs

```diff
@@ -439,70 +552,121 @@ async fn wait_for_single_account_sequence(
 /// will be, since not all of the submitted transactions were accepted.
 /// TODO, investigate whether this behaviour is desirable.
 async fn wait_for_accounts_sequence(
+    start_time: Instant,
     client: &RestClient,
     accounts: &mut [LocalAccount],
+    transactions_per_account: usize,
     wait_timeout: Duration,
+    fetch_only_once: bool,
     rng: &mut StdRng,
-) -> Result<(), HashSet<AccountAddress>> {
-    let deadline = Instant::now() + wait_timeout;
-    let addresses: Vec<_> = accounts.iter().map(|d| d.address()).collect();
-    let mut uncommitted = addresses.clone().into_iter().collect::<HashSet<_>>();
-
-    // Choose a random account and wait for its sequence number to be up to date. After that, we can
-    // query the all the accounts. This will help us ensure we don't hammer the REST API with too many
-    // query for all the accounts.
-    let account = accounts.choose(rng).expect("accounts can't be empty");
-    if wait_for_single_account_sequence(client, account, wait_timeout)
-        .await
-        .is_err()
-    {
-        return Err(uncommitted);
-    }
+) -> Result<(), HashMap<AccountAddress, usize>> {
```

Comment:
returning counts for how many transactions are missing, so that we can compute the stats correctly